src/gtc/dace/expansion.py

```diff
@@ -163,6 +163,19 @@ def visit_MaskStmt(self, node: oir.MaskStmt, **kwargs):
         body_code = [indent + b for b in body_code]
         return "\n".join([mask_str] + body_code)
 
+    def visit_While(self, node: oir.While, **kwargs):
+        body = self.visit(node.body, **kwargs)
+        cond = self.visit(node.cond, is_target=False, **kwargs)
+
+        init = "num_iter = 0"
+        max_iter = 10000
+        cond += f" and (num_iter < {max_iter})"
+        body.append("num_iter += 1")
```

Comment:
Curious: why is this needed?

Reply:
This is needed because DaCe-orchestrated mode can only run without halo exchanges at this time, and without halo exchanges, the `lagrangian_contributions` in `fv3core` will loop indefinitely.

Reply:
Sounds good. Would it be OK with you if we moved this to another branch for the time being, so that we don't mix fv3core and gt4py concerns?